framework/src/play/deps/DependenciesManager.java

```diff
@@ -183,36 +203,52 @@ public boolean problems() {
                     } else {
                         if (isPlayModule(artifact) || !isFrameworkLocal(artifact)) {
                             artifacts.add(artifact);
-                            
+
                             // Save the order of module
-                            if(isPlayModule(artifact)){
+                            if (isPlayModule(artifact)) {
                                 String mName = artifact.getLocalFile().getName();
                                 if (mName.endsWith(".jar") || mName.endsWith(".zip")) {
-                                    mName = mName.substring(0, mName.length() - 4); 
+                                    mName = mName.substring(0, mName.length() - 4);
                                 }
                             }
                         }
                     }
                 }
+            } else if (node.hasProblem() && !node.isCompletelyEvicted()) {
+                problems.add(node);
             }
         }
-        
+
         // Create directory if not exist
         File modulesDir = new File(application, "modules");
-        if(!modulesDir.exists()){
+        if (!modulesDir.exists()) {
             modulesDir.mkdir();
         }
-          
-     
-        if (!missing.isEmpty()) {
+
+        if (!missing.isEmpty() || !problems.isEmpty()) {
             System.out.println("~");
             System.out.println("~ WARNING: Some dependencies could not be downloaded (use --verbose for details),");
             System.out.println("~");
             for (ArtifactDownloadReport d : missing) {
-                String msg = d.getArtifact().getModuleRevisionId().getOrganisation() + "->" + d.getArtifact().getModuleRevisionId().getName() + " " + d.getArtifact().getModuleRevisionId().getRevision() + ": " + d.getDownloadDetails();
-                System.out.println("~\t" + msg);
+                StringBuilder msg = new StringBuilder(d.getArtifact().getModuleRevisionId().getOrganisation()).append(" -> ")
```

Comment:
We actually don't need StringBuilder here. Concatenating String with `+` is more readable, and works the same.